### PR TITLE
[doc/zh]: translate the last two parts of sparse-primary-index Chinese doc

### PR DESCRIPTION
**Pull request by selfuppen**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
1.  translate the last two part of sparse-primary-index Chinese doc.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by selfuppen** [comment]
@hanfei1991 Hi.
this commit is for the [issue](https://github.com/ClickHouse/ClickHouse/issues/53254) of the part 3,
'the English document by the two parts, Ordering key columns efficiently and Identifying single rows efficiently.'

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 22a84551f2139fa658ae2110224af8c073c5bd33 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53612/22a84551f2139fa658ae2110224af8c073c5bd33/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/zh/guides/improving-query-performance/sparse-primary-indexes.md [hunk 1]
```diff
@@ -1171,10 +1171,267 @@ ClickHouse服务器日志文件中跟踪日志确认了ClickHouse正在对索引
...
+
+虽然两个表中存储的数据完全相同（我们在两个表中插入了相同的 887 万行），但复合主键中关键字列的顺序对表的 [列数据文件](#数据按照主键排序存储在磁盘上)中的 <a href="https://clickhouse.com/docs/en/introduction/distinctive-features/#data-compression" target="_blank">压缩</a>数据所需的磁盘空间有很大影响：
+- 在具有复合主键`(URL, UserID, IsRobot)` 的表 `hits_URL_UserID_IsRobot` 中，我们按照键列的基数降序排列，此时 `UserID.bin` 数据文件占用**11.24MB**的磁盘空间。
+- 在具有复合主键`(IsRobot, UserID, URL)` 的表 `hits_IsRobot_UserID_URL` 中，我们按照键列的基数升序排列，`UserID.bin` 数据文件仅占用**877.47 KiB**的磁盘空间。
+
+对磁盘上表的列数据进行良好的压缩比不仅能节省磁盘空间，还能使需要从该列读取数据的查询（尤其是分析查询）更快，因为将列数据从磁盘移动到主内存（操作系统的文件缓存）所需的 i/o 更少。
+
+下面我们将说明，为什么主键列按基数升序排列有利于提高表列的压缩率。
+
+下图描绘了主键的磁盘上行顺序，其中键列是按基数升序排列的：
```
Comment (hanfei1991) [inline docs/zh/guides/improving-query-performance/sparse-primary-indexes.md:1363]:
```suggestion
下图阐述了主键的磁盘上行顺序，其中键列是按基数升序排列的：
```

File: docs/zh/guides/improving-query-performance/sparse-primary-indexes.md [hunk 2]
```diff
@@ -1171,10 +1171,267 @@ ClickHouse服务器日志文件中跟踪日志确认了ClickHouse正在对索引
...
+<img src={require('../../../en/guides/best-practices/images/sparse-primary-indexes-14a.png').default} class="image"/>
+
+我们讨论过 [表的行数据按主键列有序存储在磁盘上](#数据按照主键排序存储在磁盘上)。
+
+在上图中，表格的行（它们在磁盘上的列值）首先按其 `cl` 值排序，具有相同 `cl` 值的行按其 `ch` 值排序。由于第一键列 `cl` 的基数较低，因此很可能存在具有相同 `cl` 值的行。因此，`ch`值也很可能是有序的（局部地--对于具有相同`cl`值的行而言）。
+
+如果在一列中，相似的数据被放在彼此相近的位置，例如通过排序，那么这些数据将得到更好的压缩。
+一般来说，压缩算法会受益于数据的运行长度（看到的数据越多，压缩效果越好）和局部性（数据越相似，压缩率越高）。
+
+与上图不同的是，下图勾画了主键的磁盘上行顺序，其中主键列是按基数降序排列的：
```
Comment (hanfei1991) [inline docs/zh/guides/improving-query-performance/sparse-primary-indexes.md:1373]:
```suggestion
与上图不同的是，下图阐述了主键的磁盘上行顺序，其中主键列是按基数降序排列的：
```

File: docs/zh/guides/improving-query-performance/sparse-primary-indexes.md [hunk 3]
```diff
@@ -1171,10 +1171,267 @@ ClickHouse服务器日志文件中跟踪日志确认了ClickHouse正在对索引
...
+
+下图描绘了主键的磁盘上行顺序，其中键列是按基数升序排列的：
+<img src={require('../../../en/guides/best-practices/images/sparse-primary-indexes-14a.png').default} class="image"/>
+
+我们讨论过 [表的行数据按主键列有序存储在磁盘上](#数据按照主键排序存储在磁盘上)。
+
+在上图中，表格的行（它们在磁盘上的列值）首先按其 `cl` 值排序，具有相同 `cl` 值的行按其 `ch` 值排序。由于第一键列 `cl` 的基数较低，因此很可能存在具有相同 `cl` 值的行。因此，`ch`值也很可能是有序的（局部地--对于具有相同`cl`值的行而言）。
+
+如果在一列中，相似的数据被放在彼此相近的位置，例如通过排序，那么这些数据将得到更好的压缩。
+一般来说，压缩算法会受益于数据的运行长度（看到的数据越多，压缩效果越好）和局部性（数据越相似，压缩率越高）。
```
Comment (hanfei1991) [inline docs/zh/guides/improving-query-performance/sparse-primary-indexes.md:1371]:
```suggestion
一般来说，压缩算法会受益于数据的运行长度（可见的数据越多，压缩效果越好）和局部性（数据越相似，压缩率越高）。
```

File: docs/zh/guides/improving-query-performance/sparse-primary-indexes.md [hunk 4]
```diff
@@ -1171,10 +1171,267 @@ ClickHouse服务器日志文件中跟踪日志确认了ClickHouse正在对索引
...
+
+为了实现最快的检索，UUID 列[需要成为主键列](#主索引被用来选择颗粒)。
+
+我们讨论过，由于[ClickHouse 表的行数据是按主键列顺序存储在磁盘上的](#数据按照主键排序存储在磁盘上)，因此在主键或复合主键中，在基数较小的列之前设置基数非常大的列（如 UUID 列）[不利于其他表列的压缩率](#数据文件的最佳压缩率)。
+
+在最快检索速度和最佳数据压缩之间的折中方法是使用某种复合主键，其中 UUID 是最后一列关键字，位于（更）小基数关键字列之后，这些关键字列用于确保表中某些列的良好压缩比。
+
+### 一个具体例子
+
+一个具体的例子是 Alexey Milovidov 开发的明文粘贴服务 https://pastila.nl， 相关[博客](https://clickhouse.com/blog/building-a-paste-service-with-clickhouse/)。
```
Comment (hanfei1991) [inline docs/zh/guides/improving-query-performance/sparse-primary-indexes.md:1407]:
```suggestion
一个具体的例子是 Alexey Milovidov 开发的文本粘贴服务 https://pastila.nl， 相关[博客](https://clickhouse.com/blog/building-a-paste-service-with-clickhouse/)。
```

File: docs/zh/guides/improving-query-performance/sparse-primary-indexes.md [hunk 5]
```diff
@@ -1171,10 +1171,267 @@ ClickHouse服务器日志文件中跟踪日志确认了ClickHouse正在对索引
...
+- 表数据文件的压缩率。
+
+为了演示这一点，我们将使用我们的[网络流量样本数据集(web traffic sample data set)](#数据集)这个版本，
+其中每一行包含三列，分别表示互联网用户(`UserID` 列）对 URL（`URL`列）的访问是否被标记为僵尸流量（`IsRobot` 列）。
+
+我们将使用一个包含上述所有三列的复合主键，该主键可用于加快计算以下内容的典型网络分析查询速度
+- 特定 URL 有多少（百分比）流量来自机器人，或
+- 我们对特定用户是否为僵尸用户有多大把握（假定来自该用户的流量中有多大比例为僵尸流量
+
+我们使用该查询来计算我们要用作复合主键中键列的三列的基数（注意，我们使用 [URL 表函数](/docs/en/sql-reference/table-functions/url.md) 来即席查询 TSV 数据，而无需创建本地表）。在 `clickhouse client`中运行此查询：
```
Comment (hanfei1991) [inline docs/zh/guides/improving-query-performance/sparse-primary-indexes.md:1190]:
```suggestion
我们使用该查询来计算我们要用作复合主键中三个列的基数（注意，我们使用 [URL 表函数](/docs/en/sql-reference/table-functions/url.md) 来即席查询 TSV 数据，而无需创建本地表）。在 `clickhouse client`中运行此查询：
```

File: docs/zh/guides/improving-query-performance/sparse-primary-indexes.md [hunk 6]
```diff
@@ -1171,10 +1171,267 @@ ClickHouse服务器日志文件中跟踪日志确认了ClickHouse正在对索引
...
+在复合主键中，键列的顺序会对以下两方面产生重大影响：
+- 查询中过滤次关键字列的效率，以及
+- 表数据文件的压缩率。
+
+为了演示这一点，我们将使用我们的[网络流量样本数据集(web traffic sample data set)](#数据集)这个版本，
+其中每一行包含三列，分别表示互联网用户(`UserID` 列）对 URL（`URL`列）的访问是否被标记为僵尸流量（`IsRobot` 列）。
+
+我们将使用一个包含上述所有三列的复合主键，该主键可用于加快计算以下内容的典型网络分析查询速度
+- 特定 URL 有多少（百分比）流量来自机器人，或
+- 我们对特定用户是否为僵尸用户有多大把握（假定来自该用户的流量中有多大比例为僵尸流量
```
[the remark this answers is not in the document]
Comment (selfuppen) [reply docs/zh/guides/improving-query-performance/sparse-primary-indexes.md]: done

File: docs/zh/guides/improving-query-performance/sparse-primary-indexes.md (hunk 5 above)
Comment (selfuppen) [reply docs/zh/guides/improving-query-performance/sparse-primary-indexes.md]: all suggestions above have done.

**Review by hanfei1991 — APPROVED** [review]
Good job!

**Pull request merged**